coala_quickstart/generation/Bears.py

```diff
@@ -418,3 +429,52 @@ def prompt_to_activate(bear, printer):
         return False
     else:
         return prompt_to_activate(bear, printer)
+
+
+def ask_to_select_capabilties(all_capabilities,
+                              default_capabilities,
+                              console_printer):
+    """
+    Asks the users to select capabilties out of all_capabilities.
+    """
+    all_capabilities = sorted(all_capabilities)
+    indentation = " " * 4
+    PROMPT_QUESTION = ("What would you like the bears to detect or fix? "
+                       "Please select some bear capabilities using "
+                       "their numbers:")
+    console_printer.print(PROMPT_QUESTION, color="yellow")
+
+    for idx, cap in enumerate(all_capabilities):
+        color = 'cyan'
+        if cap in default_capabilities:
+            color = 'green'
+        console_printer.print(
+            indentation + str(idx + 1) + '.' + cap, color=color)
+    console_printer.print(
+        indentation + str(idx + 2) + '.' + "Select all default capabilities " +
+        "(highlighted in green)",
+        color="cyan")
+
+    selected_numbers = []
+    try:
+        selected_numbers = re.split("\D+", input())
+    except Exception:
+        # Parsing failed, choose all the default capabilities
+        selected_numbers = [idx + 2]
+
+    selected_capabilties = []
+    for idx in selected_numbers:
+        try:
+            selected_capabilties.append(all_capabilities[int(idx)-1])
```

Comment:
this looks a little hack-y

I'd rather you check if `idx` is valid and re-prompt the user (if it's <= 0 or >= len+2)

While you're printing all the options, create a set and append the indices of everything that's a default option. And then when you finally break out of that loop, check if it's the last one. If it is, use the built set as the response.